src/test/java/org/zalando/jackson/datatype/money/MonetaryAmountSerializerTest.java

```diff
@@ -228,4 +229,19 @@ public void shouldSerializeWithType() throws JsonProcessingException {
         assertThat(actual, is(expected));
     }
 
+    @Test
+    public void shouldSerializeWithCorrectLocale() throws JsonProcessingException {
+        final ObjectMapper unit = unit(module().withDefaultFormatting().withLocaleProvider(new LocaleProvider() {
+            @Override
+            public Locale getCurrentLocale(SerializerProvider serializerProvider) {
+                return Locale.UK;
+            }
+        })).enableDefaultTyping();
+
+        final String expected = "{\"amount\":{\"amount\":29.95,\"currency\":\"EUR\",\"formatted\":\"EUR29.95\"}}";
```

Comment:
Your sample is actually the default. I'd suggest to use `Locale.DE` and `"29,95 EUR"`.

Reply:
Or LocaleProvider can be mocked and `getCurrentLocale` verified